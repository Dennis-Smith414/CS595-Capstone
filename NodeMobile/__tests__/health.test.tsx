import React from 'react';
import { Text } from 'react-native';
import { render, screen } from '@testing-library/react-native';

function HealthBadge({ data }: { data?: { ok?: boolean } }) {
  return <Text>{data?.ok ? 'backend ok' : 'backend not ok'}</Text>;
}

test('renders backend ok when health.ok = true', () => {
  const { toJSON } = render(<HealthBadge data={{ ok: true }} />);
  expect(screen.getByText('backend ok')).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});
