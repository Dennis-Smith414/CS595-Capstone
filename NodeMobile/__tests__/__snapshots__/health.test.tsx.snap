// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders backend ok when health.ok = true 1`] = `
<Text>
  backend ok
</Text>
`;
